hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -106,25 +107,26 @@ public HoodieFileGroupReader(HoodieReaderContext<T> readerContext, HoodieStorage
       Option<InternalSchema> internalSchemaOpt, HoodieTableMetaClient hoodieTableMetaClient,
       TypedProperties props,
       long start, long length, boolean shouldUseRecordPosition) {
-    this(readerContext, storage, tablePath, latestCommitTime, fileSlice, dataSchema,
+    this(readerContext, storage, tablePath, latestCommitTime, dataSchema,
         requestedSchema, internalSchemaOpt, hoodieTableMetaClient, props, start, length,
-        shouldUseRecordPosition, false, false, false);
+        shouldUseRecordPosition, false, false, false, fileSlice.getBaseFile(), fileSlice.getLogFiles(), fileSlice.getPartitionPath());
   }
 
   private HoodieFileGroupReader(HoodieReaderContext<T> readerContext, HoodieStorage storage, String tablePath,
-                                String latestCommitTime, FileSlice fileSlice, Schema dataSchema, Schema requestedSchema,
+                                String latestCommitTime, Schema dataSchema, Schema requestedSchema,
                                 Option<InternalSchema> internalSchemaOpt, HoodieTableMetaClient hoodieTableMetaClient, TypedProperties props,
-                                long start, long length, boolean shouldUseRecordPosition, boolean allowInflightInstants, boolean emitDelete, boolean sortOutput) {
+                                long start, long length, boolean shouldUseRecordPosition, boolean allowInflightInstants, boolean emitDelete, boolean sortOutput,
+                                Option<HoodieBaseFile> baseFileOption, Stream<HoodieLogFile> logFiles, String partitionPath) {
```

Comment:
The params is super large now, can we have an abstraction like `HoodieInputSplit` to encapsulate these three: `Option<HoodieBaseFile> baseFileOption, Stream<HoodieLogFile> logFiles, String partitionPath`. Probably also these two: `long start, long length`

Reply:
Yes, sounds good. For the `start` and `length` does `FileRange` sound like a good name for the class?